src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -221,7 +221,12 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
     // other than the debug markers then.
     {
       if (command_buffer_to_state_.contains(command_buffer)) {
-        ERROR("Calling vkBeginCommandBuffer on a command buffer that is not in the initial state.");
+        if (!logged_begin_when_not_initial_) {
```

Comment:
Can you move this out into a function? I think it becomes easier to read if the call here (and elsewhere) is:

LogErrorOnce("error string", &logged_being_when_not_initial);

The code is already quite nested as-is.

Of course, an actual ERROR_ONCE macro would be preferable, but probably out-of-scope for this change, and I'm not sure this can be done with performance that is appropriate here (as you need to keep track of messages that have been logged and check if something has been logged before).

Reply:
As @beckerhe pointed out this can be done with a static bool per macro invocation. So I think for the use case here, we should do that.

Reply:
So I should do that right away in this PR? Can do that tomorrow.

Reply:
Done.